### ISSUE TITLE
Decimal QTY handling within bundled products

### ISSUE THREAD

**Issue by honza-pavlik** [magento/magento2#19307]
### Preconditions (*)
1. Magento 2.2.6
2. PHP 7.0.31

### Steps to reproduce (*)
1. Use decimal qty feature
2. Create bundled product containing products with decimal qty eg. 1.05 m2

### Expected result (*)
1. in case of simple product if I use decimal qty, order is placed with decimal qty. If order is changed via administrator qty remains in decimal form and it is not rounded
2. Same should be for bundled products

### Actual result (*)
1. If bundle product with decimal qty is ordered, bundled products qty are rounded
2. If received order is edited via admin, qty is rounded
3. QTY in order confirmation email are rounded (only bundled products)

**Comment by magento-engcom-team** [comment]
Hi @honza-pavlik. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@honza-pavlik do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-andrii. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-andrii Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96588`, `MAGETWO-96589` were created

**Comment by magento-engcom-team** [comment]
Hi @saphaljha. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by saphaljha** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by saphaljha** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @saphaljha. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @saphaljha, here is your [Magento instance](https://i-19307-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-19307-2-3-develop.instances.magento-community.engineering/admin](https://i-19307-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by honza-pavlik** [comment]
Hi all,
is there any progress? Plans?

**Comment by engcom-Charlie** [comment]
Hello @honza-pavlik 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates `2.3-develop` branch by provided steps.

We are closing this issue due to _branch `2.2-develop` was closed and Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3._
 See [Accepted pull requests and ported code](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules) for more details

If you still faced this issue on `2.3` please create a new Issue with all required details according to [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)
